spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -101,15 +118,50 @@ private void validateAndSetFilterParts() {
 
 	@Override
 	public Object execute(Object[] parameters) {
+		Class<?> returnedObjectType = getQueryMethod().getReturnedObjectType();
+		Object result;
+		if (isSliceQuery() || isPageQuery()) {
+			List<?> resultEntries = (List) execute(parameters, returnedObjectType, List.class, false);
+			Long totalCount = (Long) execute(parameters, Long.class, null, true);
+
+			ParameterAccessor paramAccessor =
+					new ParametersParameterAccessor(getQueryMethod().getParameters(), parameters);
+
+			Pageable pageable = paramAccessor.getPageable();
+			if (getQueryMethod().isSliceQuery()) {
```

Comment:
Could you call the convenience method here too ? `isSliceQuery()` that you use in the outer if-statement.